src/plugins/main.js

```diff
@@ -161,13 +167,13 @@ export default async (context) => {
     await syncVuex(store, newLocale, app.i18n.getLocaleMessage(newLocale))
 
     if (!initialSetup && strategy !== STRATEGIES.NO_PREFIX) {
+      const redirectPath = app.switchLocalePath(newLocale) || app.localePath('index', newLocale)
```

Comment:
This one line replaces the previous two lines. Basically switched to using `switchLocalePath` as we can now use it from this context thanks to the other fix that made it work in plugins.